lib/testutils/src/lib.rs

```diff
@@ -142,6 +142,12 @@ impl TestEnvironment {
         }
     }
 
+    /// Panic if `CI` environment variable is set to a non-empty value
+    pub fn ensure_running_outside_ci(reason: &str) {
+        let running_in_ci = std::env::var("CI").is_ok_and(|value| !value.is_empty());
+        assert!(!running_in_ci, "Running in CI, {reason}.");
```

Comment:
nit: I think it's better to add `JJ_TEST_FORCE_GPG` or something. It might make sense for down-stream packagers to enable this flag.

Reply:
I didn't follow the situation that you have in mind. My first thought is that if we create such a variable, this could be changed to fail if either CI or JJ_GPG_... is defined.

Or do you want us to use some variable we manually set in our CI workflows?

Reply:
It's just about naming. I think explicitly-named variable is better than `CI`.

Reply:
CI is kind of a standard. I think most CI runners set it by default, I've seen it in GitHub and Sourcehut, I think.


Also, do we have a need of separate variables for `taplo` and `gpg`?

Reply:
> CI is kind of a standard. I think most CI runners set it by default

Oh, I didn't know that and assumed we would set `CI=1` somewhere.

> do we have a need of separate variables for taplo and gpg?

Downstream packagers might want to forcibly enable `gpg` tests, but not `taplo`? That wouldn't be a big deal if `CI` is the standard way to handle this kind of things, though.

Reply:
It's less well-documented than I hoped. The best I found so far is https://github.com/actions/runner/pull/374/files.

There are also crates like https://github.com/zkat/is_ci, https://crates.io/crates/ci_info, <https://docs.rs/built/0.7.7/built/util/fn.detect_ci.html>. Might be a bit of an overkill.

Reply:
My current plan is to go with CI, but add a comment explaining it. Let me know if you prefer something else.

In my eyes, the main problem with this is that it's not obvious whether, if tests succeeded, they ran in CI mode or not. I'm not sure whether there's a simple solution to this, though.

An alternative idea I like in principle would be to run tests with something like `--feature ci`. However, I'm not familiar enough with Cargo to know whether features that are relevant only for tests are supported or a good idea.

Reply:
Emily found some docs: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#default-environment-variables

I think I'll put that link in a comment. Emily also says that merging this would have helped #6058, so perhaps we should merge this.